### ISSUE TITLE
imdocker: unit tests do not run and cause testbench failure

### ISSUE THREAD

**Issue by rgerhards** [rsyslog/rsyslog#3596]
#3330 introduced imdocker plus tests. Unfortunately only after merge it appears that `imdocker-unittests` does not work because they seem to be disabled. This causes rsyslog to hang (and the test invalidly return SUCCESS) and that in turn aborts other tests because of the core file.

@n2yen can you please check and suggest how we can proceed? PR would be best.

For the time being, I'll disable the test, but that really is not a good thing.

**Comment by n2yen** [comment]
@rgerhards I will have a look and produce a pr. I’m actually not sure if the specific test ‘imdocker-unittests’ is still needed as imdocker-basic and imdocker-long-logline do a better job.

**Comment by rgerhards** [comment]
> if the specific test ‘imdocker-unittests’ is still needed as imdocker-basic and imdocker-long-logline do a better job.

If that is the case, we may consider removing the unit test plumbing completely - don't keep no longer used things inside code, they'll bite sooner or later ;-)

**Comment by n2yen** [comment]
Yes, agreed.

**Comment by rgerhards** [comment]
I close this issue because the imdocker test situation described here is no longer current.

Current `tests/Makefile.am` registers the imdocker tests under the `ENABLE_IMDOCKER_TESTS` conditional, including `imdocker-basic.sh`, `imdocker-image-name.sh`, `imdocker-long-logline.sh`, `imdocker-long-options.sh`, `imdocker-new-logs-from-start.sh`, `imdocker-multi-line.sh`, and the matching vg wrappers. `tests/diag.sh` also has explicit imdocker test gating.

If there is a current imdocker test failure, it should be tracked as a new issue with the current configure flags and failing test output.
